src/System.Memory/ref/System.Memory.cs

```diff
@@ -112,6 +112,24 @@ public static partial class MemoryExtensions
 }
 namespace System.Buffers
 {
+    public sealed partial class ArrayBufferWriter : System.Buffers.IBufferWriter<byte>, System.IDisposable
+    {
+        public ArrayBufferWriter() { }
+        public ArrayBufferWriter(int initialCapacity) { }
+        public int BytesAvailable { get { throw null; } }
+        public int BytesWritten { get { throw null; } }
+        public int Capacity { get { throw null; } }
+        public System.ReadOnlyMemory<byte> OutputAsMemory { get { throw null; } }
+        public System.ReadOnlySpan<byte> OutputAsSpan { get { throw null; } }
+        public long TotalBytesWritten { get { throw null; } }
+        public void CopyToAndReset(System.IO.Stream stream) { }
+        public System.Threading.Tasks.Task CopyToAndResetAsync(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public void Dispose() { }
+        public void Reset() { }
+        void System.Buffers.IBufferWriter<System.Byte>.Advance(int count) { }
```

Comment:
I'm not sure how I feel about this being hidden. It means I can' t new it up and use it like an IBufferWriter without going through the interface.

Reply:
> Do you have a scenario that requires using these APIs directly?

All of them. The type is literally useless unless you cast it to the interface to use these methods, as there's no other way of writing to the writer. This decision doesn't make sense to me.